### PR TITLE
General setup overhaul

### PR DESCRIPTION
**Pull request by tvoigtlaender**
Overhaul of the KingMaker setup and environments
- Update environments
- Rework docker image workflow
- remove clutter from setup.sh
- Drop support for Centos8 and Ubuntu2004
- Clean up unused files
- Bugfixes

TODO:

- [x] Change back default tag of used docker images (used for testing)
- [x] Switch to main branch in submodules after they are merged
- [x] Rename envs to more meaningful names
- [ ] Set up new /cvmfs envs for local use
- [x] Figure out bug with AlmaLinux9 setup
- [x] Update documentation

### CODE REVIEW

File: processor/framework.py [hunk 1]
```diff
@@ -330,21 +353,17 @@ def get_submission_os(self):
...
-            elif os_version[0:2] == "22":
+            if os_version[0:2] == "22":
                 image_name = "ubuntu2204"
         else:
             raise Exception(
-                f"Unknown OS {distro} {os_version}, CROWN will not run without changes"
+                f"Unknown OS {distro} {os_version}, KingMaker will not run without changes"
             )
-        image = f"ghcr.io/kit-cms/kingmaker-images-{image_name}-{str(self.ENV_NAME).lower()}:main"
+        image = f"ghcr.io/kit-cms/kingmaker-images-{image_name}-{str(self.ENV_NAME).lower()}:update_env_PR"
```
Comment (harrypuuter) [inline processor/framework.py:366]: to be switched to main when merging

**Review by harrypuuter — APPROVED** [review]
looks good to me!

File: scripts/os-version.sh
[the remark this answers is not in the document]
Comment (tvoigtlaender) [reply scripts/os-version.sh:1]: Gets used in setup.sh to check for validity of the OS (https://github.com/KIT-CMS/KingMaker/blob/4875acbd52a396e95ce127c04d41fc0b27cbf032/setup.sh#L39)

File: processor/framework.py (hunk 1 above)
Comment (tvoigtlaender) [reply processor/framework.py:366]: This should only be changed back after the related kingmaker-images PR is done

Comment (tvoigtlaender) [reply processor/framework.py]: Covered by https://github.com/KIT-CMS/KingMaker/commit/7071854b5c9d885bbebdb79d24897ecf3cebf6d0

**Comment by tvoigtlaender** [comment]
Local environments under /cvmfs/etp.kit.edu are not yet updated, and the alternative setup will be applied until it is.
This does not influence the code of this PR, as the env will be placed at the predefined path in the setup.
Proper setup for versioning of cvmfs setup remains to be discussed.

**Comment by tvoigtlaender** [comment]
Waiting for https://github.com/KIT-CMS/kingmaker-images/pull/8 to be merged to fix issue of cached docker images.

**Pull request merged**